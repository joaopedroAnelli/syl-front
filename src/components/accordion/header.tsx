import { FC, PropsWithChildren } from 'react';

export type HeaderProps = PropsWithChildren<{}>;

export const Header: FC<HeaderProps> = ({ children }) => {
  return <div className='flex items-center justify-between'>{children}</div>;
};
