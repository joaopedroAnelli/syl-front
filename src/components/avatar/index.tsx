import { FC } from 'react';
import Image, { ImageProps } from 'next/image';

export const Avatar: FC<ImageProps> = ({
  alt,
  width,
  height,
  className,
  ...props
}) => {
  return (
    <Image
      {...props}
      alt={alt}
      width={width || 80}
      height={height || 80}
      className={`rounded-full ${className}`}
    />
  );
};
