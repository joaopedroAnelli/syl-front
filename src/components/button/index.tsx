import React, { ButtonHTMLAttributes } from 'react';
import { buttonVariant, ButtonVariantProps } from '@/styles/button';

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> &
  ButtonVariantProps;

export const Button: React.FC<ButtonProps> = ({
  children,
  buttonType = 'default',
  fluid,
  ...props
}) => {
  return (
    <button
      {...props}
      className={buttonVariant({
        class: props.className,
        buttonType,
        fluid,
      })}
    >
      {children}
    </button>
  );
};
