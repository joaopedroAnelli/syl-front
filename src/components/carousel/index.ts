import { Root } from './root';
import { CarouselItem } from './item';
import { NextTrigger } from './trigger/next';
import { PrevTrigger } from './trigger/prev';

const Carousel = {
  Root,
  Item: CarouselItem,
  Trigger: {
    Next: NextTrigger,
    Prev: PrevTrigger,
  },
};

export default Carousel;
