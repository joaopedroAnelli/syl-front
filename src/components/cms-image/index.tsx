import Image, { ImageProps } from 'next/image';

import { FC } from 'react';

const CMSImage: FC<ImageProps> = ({ src, alt, ...rest }) => {
  const cmsUri = process.env.CMS_HOST || '';

  return <Image src={cmsUri + src} alt={alt} {...rest} />;
};

export default CMSImage;
