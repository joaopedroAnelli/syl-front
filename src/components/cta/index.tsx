import React, { ButtonHTMLAttributes } from 'react';
import { tv } from 'tailwind-variants';

export type CTAButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  buttonType?: 'primary' | 'secondary';
  fluid?: boolean;
};

const buttonVariant = tv({
  base: 'px-4 py-2 font-bold rounded-md text-lg',
  variants: {
    buttonType: {
      default: 'bg-off-white text-charcoal-grey-500',
      primary: 'bg-deep-blue text-off-white',
      secondary: 'bg-gold text-off-white',
    },
    fluid: {
      true: 'w-full',
    },
  },
  defaultVariants: {
    buttonType: 'default',
    fluid: false,
  },
});

export const CTAButton: React.FC<CTAButtonProps> = ({
  children,
  buttonType,
  fluid,
  ...props
}) => {
  return (
    <button
      {...props}
      className={buttonVariant({
        class: props.className,
        buttonType,
        fluid,
      })}
    >
      {children}
    </button>
  );
};
