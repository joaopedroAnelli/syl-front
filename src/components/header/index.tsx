import { FC, PropsWithChildren } from 'react';
import { tv } from 'tailwind-variants';

const headerStyle = tv({
  base: 'font-serif tracking-tight font-bold',
  variants: {
    size: {
      lg: 'text-5xl md:text-6xl lg:text-7xl',
      md: 'text-xl md:text-2xl lg:text-3xl',
    },
    color: {
      primary: 'text-primary',
      secondary: 'text-secondary',
      default: 'text-off-white',
    },
  },
  defaultVariants: {
    size: 'lg',
    color: 'default',
  },
});

export type HeaderProps = PropsWithChildren<
  Parameters<typeof headerStyle>[0] & {
    as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  }
>;

export const Header: FC<HeaderProps> = ({ children, as, ...variants }) => {
  const Tag = as || 'h1';

  return <Tag className={headerStyle(variants)}>{children}</Tag>;
};
