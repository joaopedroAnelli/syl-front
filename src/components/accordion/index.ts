import { Root } from './root';
import { Header } from './header';
import { Title } from './title';
import { Indicator } from './indicator';
import { Body } from './body';
import { Description } from './description';

const Accordion = {
  Root,
  Header,
  Title,
  Indicator,
  Body,
  Description,
};

export default Accordion;
