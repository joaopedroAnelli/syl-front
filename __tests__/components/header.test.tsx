import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Header } from '@/components/header';

const resizeWindow = (x: number, y: number) => {
  window.innerWidth = x;
  window.innerHeight = y;
  window.dispatchEvent(new Event('resize'));
};

describe('Header', () => {
  it('renders a h1', () => {
    render(<Header as='h1'>Test</Header>);

    const h1 = screen.getByRole('heading');

    expect(h1).toBeInTheDocument();
  });

  it('should render with children content', () => {
    render(<Header>Test</Header>);

    const h1 = screen.getByRole('heading');

    expect(h1).toHaveTextContent('Test');
  });

  it('should render with properly style', () => {
    render(
      <Header as='h1' size='lg' color='default'>
        Test
      </Header>
    );

    const h1 = screen.getByRole('heading');

    expect(h1).toHaveClass(
      'font-serif tracking-tight font-bold text-5xl md:text-6xl lg:text-7xl text-off-white'
    );
  });

  it('should render with custom class', () => {
    render(<Header className='custom-class'>Test</Header>);

    const h1 = screen.getByRole('heading');

    expect(h1).toHaveClass(
      'font-serif tracking-tight font-bold text-5xl md:text-6xl lg:text-7xl text-off-white custom-class'
    );
  });

  it('should override color', () => {
    render(<Header color='gold'>Test</Header>);

    const h1 = screen.getByRole('heading');

    expect(h1).toHaveClass(
      'font-serif font-bold text-5xl md:text-6xl lg:text-7xl tracking-tight text-gold'
    );
  });

  it('should match snapshot', () => {
    const { container } = render(<Header>Test</Header>);

    expect(container).toMatchSnapshot();
  });

  // it('should be bigger in desktop', () => {
  //   resizeWindow(1024, 768);

  //   render(<Header size='lg'>Test</Header>);

  //   const h1 = screen.getByRole('heading');

  //   expect(h1).toHaveStyle('font-size: 4.5rem');
  // });
});
