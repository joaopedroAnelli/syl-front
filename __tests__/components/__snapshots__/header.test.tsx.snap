// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAButton should match snapshot 1`] = `
<div>
  <h1
    class="font-serif tracking-tight font-bold text-5xl md:text-6xl lg:text-7xl text-off-white"
  >
    Test
  </h1>
</div>
`;
